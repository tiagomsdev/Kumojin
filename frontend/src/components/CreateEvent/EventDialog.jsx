import React from "react";
import {
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogTitle
} from "@material-ui/core";
import EventForm from "./EventForm";

function EventDialog({handleCreateEvent}) {
  const [open, setOpen] = React.useState(false);

  function handleClickOpen(){
    setOpen(true);
  };

  function handleClose(){
    setOpen(false);
  };
  
  return (
    <div>
      <Button data-testid="newEvent-button" variant="contained" onClick={handleClickOpen}>
        + Nouveau
      </Button>
      <Dialog open={open} onClose={handleClose}>
        <DialogTitle>Créer un événement</DialogTitle>
        <DialogContent data-testid="eventDialog-content">
          <EventForm handleClose={handleClose} handleCreateEvent={handleCreateEvent}/>
        </DialogContent>
        <DialogActions>
          <Button onClick={handleClose}>Cancel</Button>
        </DialogActions>
      </Dialog>
    </div>
  );
}

export default EventDialog;
