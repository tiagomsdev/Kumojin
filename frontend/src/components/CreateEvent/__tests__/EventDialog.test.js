import {render, screen, cleanup} from '@testing-library/react';
import EventDialog from "../EventDialog";
import renderer from "react-test-renderer";

test('should render EventDialog component', () => {
    render(<EventDialog/>);

    const buttonElement = screen.getByTestId('newEvent-button');

    expect(buttonElement).toBeInTheDocument();
    expect(buttonElement).toHaveTextContent('+ Nouveau');
});

test('should matches EventDialog snapshot', () =>{
    const eventDialogTree = renderer.create(<EventDialog/>).toJSON();
    expect(eventDialogTree).toMatchSnapshot();
});