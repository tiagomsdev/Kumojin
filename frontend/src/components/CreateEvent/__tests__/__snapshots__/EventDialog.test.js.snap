// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should matches EventDialog snapshot 1`] = `
<div>
  <button
    className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-sghohy-MuiButtonBase-root-MuiButton-root"
    data-testid="newEvent-button"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    + Nouveau
  </button>
</div>
`;
